robot/rospackages/src/task_handler/scripts/task_handler_server.py

```diff
@@ -2,33 +2,26 @@
 
 import rospy
 import os
-from Listener import Listener
+import time
+import sys
+from robot.rospackages.src.task_handler.scripts.Listener import Listener
 from task_handler.srv import *
 import glob
 
-current_dir = os.path.dirname(os.path.realpath(__file__)) + "/"
-mcu_control_dir = current_dir + '../../mcu_control/scripts/'
-scripts = [mcu_control_dir + "RoverNode.py", mcu_control_dir + "ArmNode.py", mcu_control_dir + "ScienceNode.py", mcu_control_dir + "PdsNode.py", current_dir + "start_ros_stream.sh"]
-running_tasks = [Listener(scripts[0], "python3"), Listener(scripts[1], "python3"), Listener(scripts[2], "python3"), Listener(scripts[3], "python3"), Listener(scripts[4], "bash", "", 1, True)]
-known_tasks = ["rover_listener", "arm_listener", "science_listener", "pds_listener", "camera_stream"]
-known_listeners = known_tasks[:-1]
-
-i = 0
-for task in running_tasks:
-    task.set_name(known_tasks[i])
-    i += 1
-
 # return the response string after calling the task handling function
 def handle_task_request(req):
     response = "\n" + handle_task(req.task, req.status, req.args)
 
     return response
 
+# process task, status and corresponding args sent from client
+# return response message
 def handle_task(task, status, args):
-
     response = "Nothing happened"
 
     global running_tasks
+    global stream_ctr
+    global local
```

Comment:
Global variables monkaS

Reply:
yes monkaS

Reply:
and what is local, do you mean isLocal? also is stream_ctr could be seen as a temp var it's not clean from the name if its the total count of the streams or not

Reply:
When you look at the function `handle_task(task, status, args)`, ok that's fine. But I don't think it should be having variables like local and stream_ctr. It's not its responsibility.

if chosen_task is "camera_stream" then it should call a function 

handle_cam_stream_task or something

Reply:
So far I have
- [x] changed `stream_ctr` to `active_stream_ctr`
- [x] changed `local` to `is_local`
- [x] refactored parts of `handle_task` into functions:
- - [x] validate_camera_stream
- - [x] start_camera_task
- - [ ] stop_camera_task

I'm not sure if the global variables can be avoided at this point since multiple functions need them. I suppose making it a class with those global variables as static fields would be a solution. Would it be fine to open a new issue for turning it into a class? There might be other implications that are going to cause it to be a rabbit hole in itself to implement.

Reply:
nothing turns me more on then a bunch of checklists

Reply:
It's hot too see checkboxes :hot_pepper: :fire: :checkered_flag: :ballot_box_with_check: